src/api/api.js

```diff
@@ -19,441 +53,322 @@ export const getMessageConversations = ({
     assignedToMeFilter,
     markedForFollowUpFilter,
     unreadFilter,
+    currentUser,
 }) => {
-    const filters = [`messageType:eq:${messageType}`]
-    typeof status !== 'undefined' && filters.push(`status:eq:${status}`)
-    typeof priority !== 'undefined' && filters.push(`priority:eq:${priority}`)
-    markedForFollowUpFilter && filters.push('followUp:eq:true')
-    unreadFilter && filters.push('read:eq:false')
-
-    return getD2Instance()
-        .then(instance => {
-            assignedToMeFilter &&
-                filters.push(`assignee.id:eq:${instance.currentUser.id}`)
-
-            return instance.Api.getApi().get(
-                `messageConversations?pageSize=${pageSize}&page=${page}${
-                    messageFilter !== '' && messageFilter !== undefined
-                        ? `&queryString=${messageFilter}`
-                        : ''
-                }`,
-                {
-                    fields: [initialMessageConversationFields],
-                    order,
-                    filter: filters,
-                }
-            )
-        })
-        .then(result => ({
-            messageConversations: result.messageConversations,
-            pager: result.pager,
-        }))
-        .catch(error => {
-            throw error
-        })
+    const filter = [`messageType:eq:${messageType}`]
+
+    if (status) {
+        filter.push(`status:eq:${status}`)
+    }
+    if (priority) {
+        filter.push(`priority:eq:${priority}`)
+    }
+    if (markedForFollowUpFilter) {
+        filter.push('followUp:eq:true')
+    }
+    if (unreadFilter) {
+        filter.push('read:eq:false')
+    }
+    if (assignedToMeFilter) {
+        filter.push(`assignee.id:eq:${currentUser.id}`)
+    }
+
+    const query = {
+        resource: 'messageConversations',
+        params: {
+            filter,
+            pageSize,
+            page,
+            fields: initialMessageConversationFields,
+            order,
+        },
+    }
+
+    if (messageFilter) {
+        query.params.queryString = messageFilter
+    }
+
+    const { messageConversations } = await engine.query({
+        messageConversations: query,
+    })
```

Comment:
In areas like this in the api module, where we're not returning the engine directly but instead returning the result, I think we should also handle the errors that could potentially occur with the query/mutation. (also goes for the other occurences for this pattern in this module)

Reply:
Why do you think this is needed? The error is already being handled in the thunk right?

Reply:
Ah there's try / catches there, I didn't consider that. If it's all handled there it should be good.